### DISCUSSION TITLE
Supported chains

### DISCUSSION THREAD

**Discussion by jasheal** [wevm/wagmi#42] [Q&A] [answered]
Really great work here on this lib. I'm refactoring an app to use this. 

Currently I'm using `useDapp` and and I can pass in the supported chains. When a user tries to an unsupported network it provides an error and returns undefined for the `account`.

How would I achieve this or similar with wagmi?

**Comment by iduuck** [comment]
Since you are able to just implement a custom provider and connector, you are free to use whatever RPC, etc you want. Have a look at the example code for a short intro into this topic: https://github.com/tmm/wagmi/blob/main/examples/next/src/pages/_app.tsx#L18-L46

**Comment by tmm** [comment, marked as answer]
Yes, in `wagmi` you define supported chains on a per connector-basic. `InjectedConnector`, `WalletConnectConnector`, etc. all have a `chains` property. For example,

```ts
const connectors = [
  WalletConnectConnector({
    chains: [chain.mainnet, chain.rinkeby],
    options: {
      ...
    },
  })
]
```

When someone connects to an unsupported network, you can use the [`useNetwork`](https://wagmi-xyz.vercel.app/docs/hooks/useNetwork#result) hook to identify that it is unsupported (`data.unsupported`) and show a prompt for switching the network.

Will consider adding `unsupported` somewhere to response of `connector.connect` as well.

**Reply by tmm** [reply]
@jasheal [added `unsupported` to result of `connector.connect`](https://github.com/tmm/wagmi/pull/56) so you'll know immediately if the chain is supported.

**Reply by jasheal** [reply]
Awesome!
